File: result/translate/transcoder/python2java/java/MAXIMUM_GAMES_PLAYED_WINNER.java
public static int f_gold ( int N ) {
  int [ ] dp = new int [ N ] ;
  dp [ 0 ] = 1 ;
  dp [ 1 ] = 2 ;
  int i = 1 ;
  while ( dp [ i ] <= N ) {
    i = i + 1 ;
    dp [ i ] = dp [ i - 1 ] + dp [ i - 2 ] ;
  }
  return ( i - 1 ) ;
}

File: result/translate/starchat/java2cpp/c++/MAXIMUM_GAMES_PLAYED_WINNER.cpp
int f_gold ( int N ) {
  vector <int> dp ( N + 1, 0 );
  dp[0] = 1;
  dp[1] = 2;
  for ( int i = 2; i <= N; i++ ) {
    dp[i] = max ( dp[i-1], dp[i-2] ) + 1;
  }
  return dp[N];
}